src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -489,6 +492,33 @@ private void doExecute() throws MojoExecutionException {
             }
         }
     }
+    
+    private void fillSensibleHeaderTypeDefaults() throws MojoExecutionException {
+        if (headerType == null) {
+            if (classPath != null && classPath.mainClass != null) {
+                if (classPath.mainClass.toLowerCase().contains("gui")) {
+                    getLog().warn("headerType not set, defaulting to \"gui\" because the main class is named " + classPath.mainClass);
+                    headerType = "gui";
+                    return;
+                }
+            }
+
+            getLog().warn("headerType not set, defaulting to \"console\"");
+            headerType = "console";
+        }
+    }
+
+    private void fillSensibleJreDefaults() throws MojoExecutionException {
+        if (jre == null) {
+            jre = new Jre();
+        }
+
+        if (jre.path == null) {
+            String pathDef = "%JAVA_HOME%;%PATH%";
+            getLog().warn("jre.path not set, defaulting to \"" + pathDef + "\"");
+            jre.path = pathDef;
+        }
```

Comment:
`jre.path` cannot be `null` if defined via `pom.xml`, the only case to have it `null` is when you create it by hand like this. I would enclose the whole logic with `if`:
```java
        if (jre == null) {
            jre = new Jre();
            String pathDef = "%JAVA_HOME%;%PATH%";
            getLog().info("jre not defined, defaulting jre & jre.path to \"" + pathDef + "\"");
            jre.path = pathDef;
        }
```

Reply:
Can we keep it just in case? Better some unnecessary checks now than surprises later